File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public static boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int originalNumber = x;
        int reversedNumber = 0;

        while (x > 0) {
            int digit = x % 10;
            reversedNumber = reversedNumber * 10 + digit;
            x /= 10;
        }
        return originalNumber == reversedNumber;
    }
    public static void main (String[] args) {
        int x1 = 121;
        System.out.println(isPalindrome(x1));

        int x2 = 121;
        System.out.println(isPalindrome(x2));

        int x3 = 10;
        System.out.println(isPalindrome(x3));
    }
}

File: 0009-palindrome-number/0009-palindrome-number.kt
class Solution {
    fun isPalindrome(x: Int): Boolean {
        if (x < 0) {
            return false
        }
        
        val originalNumber = x
        var reversedNumber = 0
        
        var num = x
        while (num > 0) {
            val digit = num % 10
            reversedNumber = reversedNumber * 10 + digit
            num /= 10
        }
        return originalNumber == reversedNumber
    }
    
    fun main() {
        val x1 = 121
        println(isPalindrome(x1))
        
        val x2 = -121
        println(isPalindrome(x2))
        
        val x3 = 10
        println(isPalindrome(x3))
    }
}